File: docs/PayAPP-연동API-oapi-0043/examples/SAMPLE_IOS/SAMPLE_IOS/payapp_ios_sample_v3/OAPI/model/data/PAPayCancelReqData.swift
//
//  PAPayCancelReqData.swift
//  Payapp_iOS_Sample_V2
//
//  Created by Jin Won Kim on 2021/11/01.
//

import ObjectMapper

class PAPayCancelReqData: Mappable {
    
    /**
    * cmd
    * 필수
    */
    var cmd:String?
    
    /**
     * 판매자 회원 아이디
     * 필수
     */
    var userid:String?

    /**
     * 연동 KEY
     * 필수
     */
    var linkkey:String?

    /**
     * 결제요청번호
     * 필수
     */
    var mul_no:String?

    /**
     * 결제요청취소 메모
     * 필수
     */
    var cancelmemo:String?


    /**
     * 입금자명
     */
    var dpname:String?


    /**
     * 결제요청취소 구분 (0:전체취소,1:부분취소)
     */
    var partcancel:String?

    /**
     * 결제요청취소 금액 (부분취소인 경우 필수)
     */
    var cancelprice:String?

    /**
     * 결제요청취소 과세 공급가액
     * ( partcancel = cancel_taxable + cancel_taxfree + cancel_vat )
     */
    var cancel_taxable:String?

    /**
     * 결제요청취소 면세 공급가액
     */
    var cancel_taxfree:String?

    /**
     * 결제요청취소 부가세
     */
    var cancel_vat:String?
    
    required init?(map: Map){}
    init(userid:String, linkkey:String, mul_no:String, cancelmemo:String){
        self.userid = userid
        self.linkkey = linkkey
        self.mul_no = mul_no
        self.cancelmemo = cancelmemo
    }
    
    func mapping(map: Map) {
        
        cmd                             <- map["cmd"]
        userid                          <- map["userid"]
        linkkey                         <- map["linkkey"]
        mul_no                          <- map["mul_no"]
        cancelmemo                      <- map["cancelmemo"]
        dpname                          <- map["dpname"]
        partcancel                      <- map["partcancel"]
        cancelprice                     <- map["cancelprice"]
        cancel_taxable                  <- map["cancel_taxable"]
        cancel_taxfree                  <- map["cancel_taxfree"]
        cancel_vat                      <- map["cancel_vat"]
        
    }
}

File: docs/PayAPP-연동API-oapi-0043/examples/SAMPLE_ANDROID/SAMPLE_ANDROID/payapp_android_sample_v3/OApi/src/main/java/com/udid/oapi/data/PAPayCancelReqData.java
package com.udid.oapi.data;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import lombok.ToString;

/**
 * Create by 김진원
 * 결제요청 취소
 */
@Getter
@Setter
@ToString
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class PAPayCancelReqData {

    /**
     * 판매자 회원 아이디
     * 필수
     */
    @NonNull
    String userid;

    /**
     * 연동 KEY
     * 필수
     */
    @NonNull
    String linkkey;

    /**
     * 결제요청번호
     * 필수
     */
    @NonNull
    String mul_no;

    /**
     * 결제요청취소 메모
     * 필수
     */
    @NonNull
    String cancelmemo;


    /**
     * 입금자명
     */
    String dpname;


    /**
     * 결제요청취소 구분 (0:전체취소,1:부분취소)
     */
    String partcancel;

    /**
     * 결제요청취소 금액 (부분취소인 경우 필수)
     */
    String cancelprice;

    /**
     * 결제요청취소 과세 공급가액
     * ( partcancel = cancel_taxable + cancel_taxfree + cancel_vat )
     */
    String cancel_taxable;

    /**
     * 결제요청취소 면세 공급가액
     */
    String cancel_taxfree;

    /**
     * 결제요청취소 부가세
     */
    String cancel_vat;


    private static PAPayCancelReqDataBuilder builder() {
        return new PAPayCancelReqDataBuilder();
    }

    /**
     * 필수 값
     * @param userid
     * @param linkkey
     * @param mul_no
     * @param cancelmemo
     * @return
     */
    public static PAPayCancelReqDataBuilder builder(String userid, String linkkey, String mul_no, String cancelmemo) {

        return PAPayCancelReqData.builder().userid(userid).linkkey(linkkey).mul_no(mul_no).cancelmemo(cancelmemo);
    }



}
